lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -331,8 +331,8 @@ void PointConditionalDistribution::update()
     // find a feasible starting point
     SobolSequence sequence(dimension);
     Point start;
-    const UnsignedInteger size = ResourceMap::GetAsUnsignedInteger("PointConditionalDistribution-RatioUniformCandidateNumber");
-    for (UnsignedInteger i = 0; i < size; ++ i)
+    const UnsignedInteger candidateNumber = ResourceMap::GetAsUnsignedInteger("PointConditionalDistribution-RatioUniformCandidateNumber");
+    for (UnsignedInteger k = 0; k < candidateNumber; ++ k)
```

Comment:
And what happen if no candidate with postive PDF is found after the fixed number of tries? No exception? It is why I opted for a while loop. It all depends on if we want to insure that the RNG is initialized, even if it takes time, or if we accept to throw because the initialization is too costly. But if we don't throw the candidate is 100% sure to be bad after the loop!